### ISSUE TITLE
Combine date and time in a timezone

### ISSUE THREAD

**Issue by randomstuff** [sdispater/pendulum#156]
The `combine()` function is:

    combine(date, time) method of builtins.type instance
        date, time -> datetime with same date and time fields

It does not seem very useful as you usually want to combine a date and a time in a specific timezone (not in UTC). There does not seem to be support for combining date and time within a timezone.

The `combine()` function could probably be extended to something like: `combine(date, time, tzinfo=None, is_dst=None)`.

**Comment by sdispater** [comment]
I don't think modifying the signature of builtin methods is a good idea since we can't know if they will change in future Python versions.

And I may be wrong but I think you can get the same behavior by using `instance()`:

```python
>>> import pendulum
>>> from datetime import datetime
>>> pendulum.instance(datetime.combine(date, time), tz='Europe/Paris')
```

**Comment by randomstuff** [comment]
Yes, that's a way to to it it not a very satisfying one :) It would be probably be worth mentioning in the doc if this is the recommended way to do it.

**Comment by nk9** [comment]
Thanks for the suggestion of using `datetime.combine()`, it helped me work out a problem I was having: https://stackoverflow.com/questions/47004673

Can't deny that `pendulum.combine(date, time, tz='Europe/Paris')` would feel even cleaner, though I understand not wanting to extend a builtin method. I hope you're able to come up with an alternative. Maybe `merge(date, time, tz=, is_dst=)`?

**Comment by pganssle** [comment]
> I don't think modifying the signature of builtin methods is a good idea since we can't know if they will change in future Python versions.

@sdispater The [the signature of `datetime.combine`](https://docs.python.org/3/library/datetime.html#datetime.datetime.combine) has already changed in Python 3.6 to add this feature, so if you switch to `Pendulum.combine(date, time, tzinfo=self.tzinfo)`, you will be effectively backporting the current `datetime` functionality.

**Comment by gazpachoking** [comment]
I think this could probably be considered closed now?
https://github.com/sdispater/pendulum/blob/f071ed370c53caada7cb974b27b890febad1d1fa/pendulum/datetime.py#L1258-L1265
